samza-core/src/main/java/org/apache/samza/clustermanager/HostAwareContainerAllocator.java

```diff
@@ -81,13 +87,27 @@ public void assignResourceRequests()  {
 
         if (expired) {
           updateExpiryMetrics(request);
-          if (resourceAvailableOnAnyHost) {
-            log.info("Request for container: {} on {} has expired. Running on ANY_HOST", request.getContainerID(), request.getPreferredHost());
-            runStreamProcessor(request, ResourceRequestState.ANY_HOST);
+
+          if (standbyContainerManager.isPresent()) {
+
+            // if standby is enabled and an alternative-anyhost-resource is available, we try to use it
+            if (resourceAvailableOnAnyHost) {
+              standbyContainerManager.get().handleExpiredResourceRequest(containerID, request, Optional.of(peekAllocatedResource(ResourceRequestState.ANY_HOST)), this, resourceRequestState);
+            } else {
+              standbyContainerManager.get().handleExpiredResourceRequest(containerID, request, Optional.empty(), this, resourceRequestState);
```

Comment:
why invoke this with an empty resource? what state are we tracking?

Reply:
Because the StandbyContainerManager has a single method to handle expired requests, if there is no alternate resource on anyHost available, we invoke it with an empty optional resource.